packages/extension/src/components/pages/BuildTransaction/BuildTransaction.tsx

```diff
@@ -0,0 +1,174 @@
+  const wallet = getCurrentWallet();
+
+  const [jsonInput, setJsonInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleBack = useCallback(() => {
+    navigate(SETTINGS_PATH);
+  }, [navigate]);
+
+  const handleSignTransaction = useCallback(() => {
+    const key = generateKey();
+    try {
+      JSON.parse(jsonInput);
+    } catch (e) {
+      setErrorMessage('Invalid JSON');
+      return;
+    }
+
+    try {
+      const parsed = JSON.parse(jsonInput);
+      if (!('Account' in parsed)) {
+        parsed.Account = wallet?.publicAddress;
+      }
+
+      validate(parsed);
+      saveInChromeSessionStorage(key, JSON.stringify({ transaction: parsed })).then(() =>
+        navigate(
+          `${SIGN_TRANSACTION_PATH}?${STORAGE_MESSAGING_KEY}=${key}&inAppCall=true&sign=transaction`
+        )
+      );
+    } catch (e) {
+      setErrorMessage('Invalid transaction');
+      return;
+    }
+    setErrorMessage('');
+  }, [jsonInput, navigate, wallet?.publicAddress]);
+
+  const handleSubmitTransaction = useCallback(() => {
+    const key = generateKey();
+    try {
+      JSON.parse(jsonInput);
+    } catch (e) {
+      setErrorMessage('Invalid JSON');
+      return;
+    }
+
+    try {
+      const parsed = JSON.parse(jsonInput);
+      if (!('Account' in parsed)) {
+        parsed.Account = wallet?.publicAddress;
+      }
+
+      validate(parsed);
+      saveInChromeSessionStorage(key, JSON.stringify({ transaction: parsed })).then(() =>
+        navigate(
+          `${SUBMIT_TRANSACTION_PATH}?${STORAGE_MESSAGING_KEY}=${key}&inAppCall=true&submit=transaction`
+        )
+      );
+    } catch (e) {
+      setErrorMessage('Invalid transaction');
+      return;
+    }
+    setErrorMessage('');
+  }, [jsonInput, navigate, wallet?.publicAddress]);
+
+  const handleBeautifyJson = useCallback(() => {
+    try {
+      const parsedJson = JSON.parse(jsonInput);
+      const beautifiedJson = JSON.stringify(parsedJson, null, 2);
+      setJsonInput(beautifiedJson);
+      setErrorMessage('');
+    } catch (e) {
+      setErrorMessage('Invalid JSON - cannot beautify');
+    }
+  }, [jsonInput]);
+
+  const jsonEditorStyle = `
+    .json-editor {
+      outline: none;
+    }`;
+
+  return (
+    <PageWithReturn title="Build Transaction" onBackClick={handleBack}>
+      <style>{jsonEditorStyle}</style>
+      <div style={{ margin: '1rem' }}>
+        <Typography variant="body1" style={{ marginBottom: '1rem' }}>
+          Please type or paste your JSON data:
+        </Typography>
+        <div style={{ maxHeight: editorHeight, overflow: 'auto' }}>
+          <Editor
+            value={jsonInput}
+            textareaClassName={'json-editor'}
+            onValueChange={(input) => setJsonInput(input)}
+            highlight={(code) => highlight(code, languages.json, 'json')}
+            padding={10}
+            style={{
+              fontFamily: '"Fira code", "Fira Mono", monospace',
```

Comment:
How do you make sure that this font family is present?

Reply:
I would say it is embedded since it's the one used in the documentation of the library